File: 31-07-2025_Bitwise_ORs_of_Subarrays.java
class Solution {
    public int subarrayBitwiseORs(int[] arr) {
        Set<Integer> res = new HashSet<>();
        Set<Integer> cur = new HashSet<>();

        for (int num : arr) {
            Set<Integer> newCur = new HashSet<>();
            newCur.add(num);
            for (int x : cur) {
                newCur.add(x | num);
            }
            cur = newCur;
            res.addAll(cur);
        }

        return res.size();
    }

}

File: 31-07-2025_Bitwise_ORs_of_Subarrays.cpp
class Solution {
public:

int subarrayBitwiseORs(vector<int>& arr) {
    unordered_set<int> res;
    unordered_set<int> cur;

    for (int num : arr) {
        unordered_set<int> new_cur = {num};
        for (int x : cur) {
            new_cur.insert(x | num);
        }
        cur = new_cur;
        for (int x : cur) {
            res.insert(x);
        }
    }

    return res.size();
}

};
